Network/HTTP/Req.hs

```diff
@@ -2009,6 +2011,22 @@ data HttpException
 
 instance Exception HttpException
 
+-- | Subset of 'HttpException' that corresponds to 'L.StatusCodeException'.
+--
+-- @since 3.12.0
+data StatusCodeException
+  = StatusCodeException IgnoreResponse ByteString
```

Comment:
The use of `IgnoreResponse` is confusing here. Try to avoid using types that are semantically wrong but just happen to have some potentially useful functions defined for them. That's a road to insanity.

Which methods of `HttpResponse` you find useful in the context of response inspection?

Reply:
I see. It was not a good idea to use `IgnoreResponse`. I think `responseStatusCode` and `responseHeader` methods are useful to inspect non-2XX responses.